src/main/java/net/buildtheearth/terraplusplus/TerraMod.java

```diff
@@ -41,29 +43,25 @@ public void construction(FMLConstructionEvent event) {
     public void preInit(FMLPreInitializationEvent event) {
         LOGGER = event.getModLog();
         EarthWorldType.create();
-
-        // This is just a handy shortcut when creating new BTE worlds on the client not needed on the server
-        // It is critical that this happens after the EarthWorldType is registered
-        if (Side.CLIENT == event.getSide()) {
-            BTEWorldType.create();
-        }
     }
 
     @EventHandler
     public void init(FMLInitializationEvent event) {
         if (TerraConfig.threeWater) {
             MinecraftForge.EVENT_BUS.register(WaterDenier.class);
         }
-        if (Side.CLIENT == event.getSide()) {
-            MinecraftForge.EVENT_BUS.register(BTEWorldType.class);
-        }
         MinecraftForge.TERRAIN_GEN_BUS.register(GenerationEventDenier.class);
 
         PermissionAPI.registerNode(TerraConstants.othersCommandNode, DefaultPermissionLevel.OP, "Allows a player to control another player in terra commands");
     }
 
     @EventHandler
     public void postInit(FMLPostInitializationEvent event) {
+    	if(event.getSide() == Side.CLIENT) {
+    		// Force the static resources to load here so the GUI doesn't freeze for 5s when first opened
```

Comment:
It also loads textures. On my laptop, opening these GUIs for the first time freezes the game for around 3 seconds.

Reply:
it *doesn't* load any textures, though.

- loading `AdvancedEarthGui` initializes two object fields:
  - `DIRECTIONS_TEXTURE`, which is a `ResourceLocation` and thus has a negligible construction cost+memory overhead
  - `SRC_CACHE`, which is a soft `Ref` and also has a negligible construction cost (the value inside isn't computed the first time until being accessed the first time)
- `PresetEarthGui` should be relatively cheap as well

Reply:
You are right it's not loading textures at that point. But what is it loading then ? Be cause I can assure you there was a very unpleasant freeze when opening those GUIs for the first time until I added that piece of code.

Reply:
I think it's just loading a lot of things because it is the first time lots of Terra++ classes get loaded. Including `DymaxionProjection`, which reads `conformal.lzma` which from my testing seems to be one of the main reasons for the freezes.